python/pyarrow/_dataset.pyx

```diff
@@ -887,6 +911,14 @@ cdef class ParquetFileFragment(FileFragment):
         FileFragment.init(self, sp)
         self.parquet_file_fragment = <CParquetFileFragment*> sp.get()
 
+    def __reduce__(self):
+        return self.format.make_fragment, (
+            self.path,
```

Comment:
I suppose you might need to do the same `self.path if buffer is None else buffer,` here as you did for `FileFragment` ?

Reply:
Fixed it, also added a test.